### PR TITLE
Fix intermittent failure in ScheduledFlowTests 

### PR DESCRIPTION
**Pull request by rick-r3**
Solve race condition across testing various counters by moving to single threaded network and testing counters on that single thread.

### CODE REVIEW

**Comment by mnesbit** [comment]
I am not keen on this. First, I have in fact used the similar test pattern in several other places. Including in CorDapps outside of the platform. Second, I am worried that we are just going back to single threading that will again hide threading issues I was starting to expose.

**Comment by rick-r3** [comment]
Now uses a global atomic activity counter for the whole MockNetwork, so no need for single threaded network.  Just checking for reliability on CI.

File: node/src/main/kotlin/net/corda/node/services/events/NodeSchedulerService.kt [hunk 1]
```diff
@@ -182,6 +179,7 @@ class NodeSchedulerService(private val database: Database,
             val scheduledLogic: FlowLogic<*>? = getScheduledLogic()
             if (scheduledLogic != null) {
                 subFlow(scheduledLogic)
+                scheduler.unfinishedSchedules.countDown()
             }
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/events/NodeSchedulerService.kt]: Yes, I agree extending the scope of the countdown till after the subFlow completes makes sense. Good catch.

**Comment by mnesbit** [comment]
Yes I like this version much more.

**Comment by rick-r3** [comment]
Failed in CI

**Comment by rick-r3** [comment]
We tracked down a race condition between counting down the flows counter and the database commit of the flow exit.  Need to count down after the flow commits.  Back in testing on CI.

**Comment by rick-r3** [comment]
Passed CI 16 times.  Merging.

**Pull request merged**